src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -639,3 +641,52 @@ ErrorIfAnyMetadataNodeOutOfSync(List *metadataNodeList)
 		}
 	}
 }
+
+
+/*
+ * IsWorkerTheCurrentNode checks if the given worker refers to the
+ * the current node by comparing the server id of the worker and of the
+ * current nodefrom pg_dist_node_metadata
+ */
+bool
+IsWorkerTheCurrentNode(WorkerNode *workerNode)
+{
+	int connectionFlags = REQUIRE_METADATA_CONNECTION;
+
+	MultiConnection *workerConnection =
+		GetNodeUserDatabaseConnection(connectionFlags,
+									  workerNode->workerName,
+									  workerNode->workerPort,
+									  CurrentUserName(),
+									  NULL);
+	const char *command =
+		"SELECT metadata ->> 'server_id' AS server_id FROM pg_dist_node_metadata";
+
+	SendRemoteCommand(workerConnection, command);
+
+	PGresult *result = GetRemoteCommandResult(workerConnection, true);
+	List *commandResult = ReadFirstColumnAsText(result);
+
+	if (commandResult == NIL)
```

Comment:
maybe be a little more strict: if(list_length(commandResult) != 1))

(not sure if otherwise is ever possible, but still)